scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java

```diff
@@ -1027,31 +1007,37 @@ public void taskRestarted(final InternalJob job, final InternalTask task, final
             public Void executeWork(Session session) {
                 long jobId = jobId(job);
 
-                String jobUpdate = "update JobData set status = :status, "
-                        + "numberOfPendingTasks = :numberOfPendingTasks, "
-                        + "numberOfRunningTasks = :numberOfRunningTasks where id = :jobId";
+                String jobUpdate = "update JobData set status = :status, " +
+                    "numberOfPendingTasks = :numberOfPendingTasks, " +
+                    "numberOfRunningTasks = :numberOfRunningTasks, " +
+                    "numberOfFailedTasks = :numberOfFailedTasks, " +
+                    "numberOfFaultyTasks = :numberOfFaultyTasks, " +
+                    "numberOfInErrorTasks = :numberOfInErrorTasks " + "where id = :jobId";
 
                 JobInfo jobInfo = job.getJobInfo();
 
-                session.createQuery(jobUpdate).setParameter("status", jobInfo.getStatus()).setParameter(
-                        "numberOfPendingTasks", jobInfo.getNumberOfPendingTasks()).setParameter(
-                        "numberOfRunningTasks", jobInfo.getNumberOfRunningTasks()).setParameter("jobId",
-                        jobId).executeUpdate();
+                session.createQuery(jobUpdate).setParameter("status", jobInfo.getStatus())
+                        .setParameter("numberOfPendingTasks", jobInfo.getNumberOfPendingTasks())
+                        .setParameter("numberOfRunningTasks", jobInfo.getNumberOfRunningTasks())
+                        .setParameter("numberOfFailedTasks", jobInfo.getNumberOfFailedTasks())
+                        .setParameter("numberOfFaultyTasks", jobInfo.getNumberOfFaultyTasks())
+                        .setParameter("numberOfInErrorTasks", jobInfo.getNumberOfInErrorTasks())
+                        .setParameter("jobId", jobId).executeUpdate();
 
                 TaskData.DBTaskId taskId = taskId(task);
 
-                String taskUpdate = "update TaskData set taskStatus = :taskStatus, "
-                        + "numberOfExecutionLeft = :numberOfExecutionLeft,"
-                        + "numberOfExecutionOnFailureLeft = :numberOfExecutionOnFailureLeft"
-                        + " where id = :taskId";
+                String taskUpdate = "update TaskData set taskStatus = :taskStatus, " +
+                    "numberOfExecutionLeft = :numberOfExecutionLeft," +
+                    "numberOfExecutionOnFailureLeft = :numberOfExecutionOnFailureLeft" +
+                    " where id = :taskId";
 
                 TaskInfo taskInfo = task.getTaskInfo();
 
                 session.createQuery(taskUpdate).setParameter("taskStatus", taskInfo.getStatus())
                         .setParameter("numberOfExecutionLeft", taskInfo.getNumberOfExecutionLeft())
                         .setParameter("numberOfExecutionOnFailureLeft",
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "numberOfExecutionOnFailureLeft" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)